src/geo/map.js

```diff
@@ -21,24 +21,78 @@ var Map = Model.extend({
     provider: 'leaflet'
   },
 
-  initialize: function() {
+  initialize: function (attrs, options) {
+    options = options || {};
     this.layers = new Layers();
+    this.geometries = new Backbone.Collection();
+
+    this._windshaftMap = options.windshaftMap;
+    this._dataviewsCollection = options._dataviewsCollection;
+
+    this._initBinds();
+  },
 
-    this.layers.bind('reset', function() {
-      if(this.layers.size() >= 1) {
+  _initBinds: function () {
+    this.layers.bind('reset', function () {
+      if (this.layers.size() >= 1) {
         this._adjustZoomtoLayer(this.layers.models[0]);
       }
     }, this);
 
-    this.layers.bind('reset', this._updateAttributions, this);
-    this.layers.bind('add', this._updateAttributions, this);
-    this.layers.bind('remove', this._updateAttributions, this);
+    // TODO: When the order of the layers change, the instance of the map needs
+    // to be re-recreated
+    this.layers.bind('reset', this._onLayersResetted, this);
+    this.layers.bind('add', this._onLayerAdded, this);
+    this.layers.bind('remove', this._onLayerRemoved, this);
     this.layers.bind('change:attribution', this._updateAttributions, this);
 
-    this.geometries = new Backbone.Collection();
+    if (this._dataviewsCollection) {
+
+      // When new dataviews are defined, a new instance of the map needs to be created
+      this._dataviewsCollection.bind('add', _.debounce(this._onDataviewAdded, 10), this);
```

Comment:
As of Backbone v1 we can use the more concise way/recommended way of listening to events:
```js
this.listenTo(otherObj, 'eventName', handler);
```

This way we can skip using the last `this` arg since the context will always be the `otherObj`. See http://backbonejs.org/#Events-listenTo

Btw, since using the debounce, you probably want to bind the context on the handler to be sure it's using the correct context, i.e. `_.debounce(this._onDataviewAdded.bind(this), 10),`